### ISSUE TITLE
SpaceAroundOperators raised when using safe navigation with an array

### ISSUE THREAD

**Issue by livedo** [bbatsov/rubocop#5750]
SpaceAroundOperators is raised when safe-navigating a square brackets method (`foo&[:bar]`).

--------

## Expected behavior

I would expect SpaceAroundOperators to ignore the safe navigation operator.

## Actual behavior

SpaceAroundOperators is raised and auto-correct produces:

```
foo & [:bar]
```

## Steps to reproduce the problem

Using the [Rails config](https://github.com/toshimaru/rubocop-rails/blob/master/config/rails.yml). The following code produces the problem:

```
foo&[:bar]
```

## RuboCop version

```
$ rubocop -V
0.54.0 (using Parser 2.5.0.5, running on ruby 2.5.1 x86_64-darwin17)
```

**Comment by Drenmi** [comment]
Well. You aren't safe navigating a bracket method. 🙂 

```ruby
foo&[:bar]
```

Is this:

```ruby
foo & [:bar]
```

What you meant to do was probably:

```ruby
foo&.[](:bar)
```

**Comment by livedo** [comment]
Uhh, momentary insanity. My bad. :p

**Comment by Smarre** [comment]
For what it is worth (and I get this in an old issue), I recently stumbled to this myself. I think `foo&.[](:bar)` is not good syntax to use, as it has much visual overhead. I stumbled to this because SafeNavigation cop wants to guard against foo["bar"], where IMHO an if is better here. But there is no way to disable the cop only for [] case as far as I see it. At least documentation makes no mention of it.
